metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -129,18 +148,29 @@ public void execute(Tuple tuple) {
       boolean ackTuple = !writer.handleAck();
       int numWritten = 0;
       if(sensorParserConfig != null) {
+        Map<String, Object> metadata = getMetadata(tuple, sensorParserConfig.readMetadata());
         List<FieldValidator> fieldValidations = getConfigurations().getFieldValidations();
         Optional<List<JSONObject>> messages = parser.parseOptional(originalMessage);
         for (JSONObject message : messages.orElse(Collections.emptyList())) {
           message.put(Constants.SENSOR_TYPE, getSensorType());
```

Comment:
* If you turn on parsing metadata and do not indicate to merge the metadata, it will be available to stellar field transformations.
* If you turn on parsing metadata and *do* indicate that you wish to merge the metadata, it will be both merged and available to stellar field transformations.